### PR TITLE
Summary file/missing cohomology lemmas

### PR DESCRIPTION
**Pull request by aljungstrom**
@mortberg

### CODE REVIEW

File: Cubical/Algebra/AbGroup/Properties.agda [hunk 1]
```diff
@@ -24,3 +35,86 @@ module AbGroupTheory (A : AbGroup ℓ) where
...
+fst (multₗHom G n) g = n ℤ[ (AbGroup→Group G) ]· g
+snd (multₗHom G n) = makeIsGroupHom (ℤ·isHom n G)
+
+-- Abelian groups quotiented by a natural number
+_/^_ : (G : AbGroup ℓ) (n : ℕ) → AbGroup ℓ
+G /^ n =
+  Group→AbGroup
+    ((AbGroup→Group G)
+     / (imSubgroup (multₗHom G (pos n))
+      , isNormalIm (multₗHom G (pos n)) (AbGroupStr.+Comm (snd G))))
```
Comment (mortberg) [inline Cubical/Algebra/AbGroup/Properties.agda:110]: Hmm, this back and forth is a bit weird. Wouldn't it be nicer to have general lemmas for quotienting AbGroups? I have a vague recollection of wanting this when defining homology groups for chain complexes... Maybe I did it then?

File: Cubical/Algebra/AbGroup/Properties.agda [hunk 2]
```diff
@@ -24,3 +35,86 @@ module AbGroupTheory (A : AbGroup ℓ) where
...
+   (SQ.elimProp2 (λ _ _ → squash/ _ _)
+     λ a b → cong [_] (AbGroupStr.+Comm (snd G) a b))
+
+-- Torsion subgrous
+_[_]ₜ : (G : AbGroup ℓ) (n : ℕ) → AbGroup ℓ
+G [ n ]ₜ =
+  Group→AbGroup (Subgroup→Group (AbGroup→Group G)
```
Comment (mortberg) [inline Cubical/Algebra/AbGroup/Properties.agda:117]: Same here... All this back and forth should be done once and for all somewhere

File: Cubical/Axiom/Choice.agda [hunk 3]
```diff
@@ -0,0 +1,58 @@
+{-# OPTIONS --safe #-}
+
```
Comment (mortberg) [inline Cubical/Axiom/Choice.agda:2]: Maybe a header would be good for this particular file summarizing what it contains?

File: Cubical/Axiom/Choice.agda [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+open import Cubical.Data.Nat.Order
+
+open import Cubical.HITs.Truncation as TR
+open import Cubical.HITs.PropositionalTruncation as PT
+
+private
+  variable
+    ℓ ℓ' ℓ'' : Level
+
+-- Stuff about choice:
```
Comment (mortberg) [inline Cubical/Axiom/Choice.agda:24]: This comment is not very helpful...

File: Cubical/Cohomology/EilenbergMacLane/Base.agda [hunk 5]
```diff
@@ -297,6 +341,23 @@ snd (coHomHom n f) =
   makeIsGroupHom (ST.elim2 (λ _ _ → isOfHLevelPath 2 squash₂ _ _)
    λ f g → refl)
 
+coHomEquiv : ∀ {ℓ''} {A : Type ℓ} {B : Type ℓ'} {G : AbGroup ℓ''}
+            (n : ℕ)
+         → (Iso A B)
+         → AbGroupEquiv (coHomGr n G B) (coHomGr n G A)
```
Comment (mortberg) [inline Cubical/Cohomology/EilenbergMacLane/Base.agda:347]: Why is A and B swapped in the conclusion?

File: Cubical/Cohomology/EilenbergMacLane/CupProduct.agda [hunk 6]
```diff
@@ -17,6 +17,7 @@ open import Cubical.Algebra.Ring
 open import Cubical.Algebra.CommRing
 open import Cubical.Algebra.CommRing.Instances.IntMod
 open import Cubical.Algebra.Group.Instances.IntMod
+open import Cubical.Algebra.AbGroup.Instances.IntMod
```
Comment (mortberg) [inline Cubical/Cohomology/EilenbergMacLane/CupProduct.agda:20]: Is this needed? (just wondering as nothing else is changed in the file...)

File: Cubical/Cohomology/EilenbergMacLane/Groups/RP2wedgeS1.agda [hunk 7]
```diff
@@ -29,6 +29,7 @@ open import Cubical.Algebra.Group.MorphismProperties
 open import Cubical.Algebra.Group.Morphisms
 open import Cubical.Algebra.AbGroup.Base
 open import Cubical.Algebra.AbGroup.Instances.Unit
+open import Cubical.Algebra.AbGroup.Instances.IntMod
```
Comment (mortberg) [inline Cubical/Cohomology/EilenbergMacLane/Groups/RP2wedgeS1.agda:32]: Import needed?

File: Cubical/Cohomology/EilenbergMacLane/Rings/KleinBottle.agda [hunk 8]
```diff
@@ -47,6 +48,7 @@ open import Cubical.Algebra.CommRing.Instances.Polynomials.MultivariatePoly-Quot
 open import Cubical.Algebra.AbGroup.TensorProduct
 open import Cubical.Algebra.Ring
 open import Cubical.Algebra.DirectSum.DirectSumHIT.Base
+open import Cubical.Algebra.AbGroup.Instances.IntMod
```
Comment (mortberg) [inline Cubical/Cohomology/EilenbergMacLane/Rings/KleinBottle.agda:51]: New imports needed?

**Review by mortberg — CHANGES_REQUESTED** [review]
Great stuff! I had some minor comments that should all be easy to fix

File: Cubical/Cohomology/EilenbergMacLane/Base.agda (hunk 5 above)
Comment (aljungstrom) [reply Cubical/Cohomology/EilenbergMacLane/Base.agda:347]: wanted to stay true to the fact that coHom is contravariant

File: Cubical/Algebra/AbGroup/Properties.agda (hunk 2 above)
Comment (aljungstrom) [reply Cubical/Algebra/AbGroup/Properties.agda:117]: Same comment as above

File: Cubical/Cohomology/EilenbergMacLane/CupProduct.agda (hunk 6 above)
Comment (aljungstrom) [reply Cubical/Cohomology/EilenbergMacLane/CupProduct.agda:20]: Yes, I moved the definition of Z/2 there, which is why it's needed in so many places now.

File: Cubical/Cohomology/EilenbergMacLane/Groups/RP2wedgeS1.agda (hunk 7 above)
Comment (aljungstrom) [reply Cubical/Cohomology/EilenbergMacLane/Groups/RP2wedgeS1.agda:32]: yes

File: Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda [hunk 9]
```diff
@@ -59,6 +59,7 @@ open import Cubical.Algebra.Group.MorphismProperties
 open import Cubical.Algebra.Group.Instances.IntMod
 open import Cubical.Algebra.AbGroup
 open import Cubical.Algebra.AbGroup.Instances.Unit
+open import Cubical.Algebra.AbGroup.Instances.IntMod
```
[the remark this answers is not in the document]
Comment (aljungstrom) [reply Cubical/Cohomology/EilenbergMacLane/Rings/RP2wedgeS1.agda:62]: yes

File: Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda [hunk 10]
```diff
@@ -17,6 +17,7 @@ open import Cubical.Algebra.DirectSum.DirectSumHIT.Base
 open import Cubical.Algebra.Ring
 open import Cubical.Algebra.CommRing
 open import Cubical.Algebra.CommRing.Instances.IntMod
+open import Cubical.Algebra.AbGroup.Instances.IntMod
```
[the remark this answers is not in the document]
Comment (aljungstrom) [reply Cubical/Cohomology/EilenbergMacLane/Rings/Z2-properties.agda:20]: yes

**Review by mortberg — APPROVED** [review]

**Comment by mortberg** [comment]
Perfect!

**Pull request merged**